### DISCUSSION TITLE
Question about aql query

### DISCUSSION THREAD

**Discussion by phamvietquang123** [eclipse-sirius/sirius-desktop#382] [Q&A] [answered]
Hi, does aql queries support Nested if-else statements ?  I can't seem to find it on acceleo documentation

**Comment by lredor** [comment]
Hi,
I found [this](https://eclipse.dev/acceleo/documentation/#Conditions) in the documentation. Doesn't this match what you're looking for?
Regards,
Laurent

**Reply by phamvietquang123** [reply]
Hi, I have read this, but it only illustrates one if-else statement. I want a nested if-else statement, though, for example if .. elseif .. else endif

**Reply by lredor** [reply, marked as answer]
OK sorry I misunderstood.
It is not documented, but it seems to be OK. For example, if you put the bellow expression in the Interpreter view, you can see that it will be OK.

`aql:if self.name.startsWith('a') then if self.name.startsWith('ab') then 'Start With ab' else 'Start With a' endif else 'Not start with a' endif`

- Case 1
![2024-04-26_12h11_21](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/1f9dfde2-ba1d-4f4c-9d69-072dbc660810)
- Case 2
![2024-04-26_12h11_26](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/97505ae1-aefd-4aed-9974-10e8d5dd5735)
- Case 3
![2024-04-26_12h11_35](https://github.com/eclipse-sirius/sirius-desktop/assets/1825882/c970166d-88b8-4b86-a827-4b0debe10422)

**Reply by phamvietquang123** [reply]
It works, thanks.
